app/bundles/EmailBundle/Helper/EmailValidator.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\Helper;
+
+use Mautic\EmailBundle\EmailEvents;
+use Mautic\EmailBundle\Event\EmailValidationEvent;
+use Mautic\EmailBundle\Exception\InvalidEmailException;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class EmailValidator.
+ */
+class EmailValidator
```

Comment:
Symfony has build-in validator for e-mail, why not to use it?
https://symfony.com/doc/current/reference/constraints/Email.html

Reply:
Mainly to add our support for plugin validation that can leverage future integrations like NeverBounce.